src/ParallelAlgorithms/LinearSolver/Schwarz/ElementCenteredSubdomainData.hpp

```diff
@@ -0,0 +1,214 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <ostream>
+#include <pup.h>
+#include <string>
+#include <tuple>
+
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/LinearSolver/InnerProduct.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace LinearSolver::Schwarz {
+
+/*!
+ * \brief Data on an element-centered subdomain
+ *
+ * An element-centered subdomain consists of a central element and overlap
+ * regions with all neighboring elements. This class holds data on such a
+ * subdomain. It supports vector space operations (addition and scalar
+ * multiplication) and an inner product, which allow to use this data type with
+ * linear solvers (see e.g. `LinearSolver::Serial::Gmres`).
+ */
+template <size_t Dim, typename TagsList>
+struct ElementCenteredSubdomainData {
+  static constexpr size_t volume_dim = Dim;
+  using ElementData = Variables<TagsList>;
+  using OverlapData = ElementData;
```

Comment:
To apply a DG operator to the subdomain data I need to pass `Variables` to our routines that apply differentiation matrices, compute fluxes and so on, and these `Variables` have to contain data on a single LGL mesh. That's why I store the `Variables` for the central element and for the overlaps with each neighbor separately. It would be nice to store all data contiguously and get "views" into this data that I can pass along as `Variables`, but I don't think that's currently possible. If I will find that this subdomain data class is relevant for performance I may have to implement something like that though.